scripts/check-test-determinism.py

```
@@ -316,6 +357,425 @@ def _is_assertion_line(line: str) -> bool:
+    literals: list[tuple[int, int]],
+) -> Optional[int]:
+    if not literals:
+        return None
+    first = line[literals[0][0] : literals[0][1]]
+    if first.rsplit("/", 1)[-1] != "env":
+        return 0
+
+    index = 1
+    while index < len(literals):
+        token = line[literals[index][0] : literals[index][1]]
+        if token == "--":
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "token"

(S105)